File: src/morse-code-error.js
/* eslint-disable max-classes-per-file */
/**
 * Error for when the corresponding morse code symbol is not found in MORSE_TO_CHAR object.
 */
class MorseCodeError extends Error {
  constructor(message) {
    super(message);
    this.name = 'MorseCodeError';
  }
}

/**
 * Error for when the corresponding character in text is not found in the CHAR_TO_MORSE object.
 */
class MorseCodeNotFound extends Error {
  constructor(message) {
    super(message);
    this.name = 'MorseCodeNotFound';
  }
}

module.exports = {
  MorseCodeError,
  MorseCodeNotFound,
};

File: src/morse_code_error.py
class MorseCodeError(Exception):
    """Throws an error if the character is not found in MORSE_TO_CHAR_DICT."""

    def __init__(self, message):
        self.message = message
        super().__init__(self.message)

class MorseCodeNotFound(Exception):
    """Throws an error if there isn't any equivalent of a char in CHAR_TO_MORSE_DICT."""

    def __init__(self, message):
        self.message = message
        super().__init__(self.message)
